R/create.R

```diff
@@ -581,13 +588,13 @@ create_initial_conditions <- function(data) {
         meanlog = data$ls_meanlog,
         sdlog = ifelse(data$ls_sdlog > 0, data$ls_sdlog * 0.1, 0.01)
       ))
-      # nolint start
-      out$rho <- ifelse(out$rho > data$ls_max, data$ls_max - 0.001,
-        ifelse(out$rho < data$ls_min, data$ls_min + 0.001,
-          out$rho
+
+      out$rho <- data.table::fcase(
+        out$rho > data$ls_max, data$ls_max - 0.001,
+        out$rho < data$ls_min, data$ls_min + 0.001,
+        default = out$rho
         )
```

Comment:
Stan expects `rho` to be an array so I think you'll need

```suggestion
      out$rho <- array(data.table::fcase(
        out$rho > data$ls_max, data$ls_max - 0.001,
        out$rho < data$ls_min, data$ls_min + 0.001,
        default = out$rho
        ))
```

If doing this you can remove the `array` a few lines above and just do `out$rho <- rlnorm`...

Reply:
Nice catch. Thanks, Seb. My knowledge of stan is too limited for me to have caught that.

Reply:
Sorry I thought I commented this but it was caught up as pending. Good spot. Generally if tests are failing checking if anything is different about the parts of the code base can be a good debug if you don't really get how they are later used. 

> this seems to be an issue when passed to stan (the dimension warning). It might be as simple as wrapping this in an array call.